### ISSUE TITLE
better understand the metrics returned during the training

### ISSUE THREAD

**Issue by user-ds** [eriklindernoren/PyTorch-YOLOv3#556]
Hello, Erik, everybody. Can someone clarify those points:

- I put all my images in labels in the directory data/custom and I specified de distribution in the in train.txt and valid.txt. When I do a >>python 3 train.py, when is my validation dataset used? Are the metrics relatives to the test dataset, validation dataset or both?
- Can you tell me a little bit more about the Layer-0, Layer-1 and Layer-2? Is it due to multiscaling? And are the grid_size, the number of pixels in each cell, or the number of cells per line and row?
- I trained it two times. Once with Multisclae_training = True and once with = False. Why do I have metrics for the three layers when multiscaling = false (deactivated)?
- Is cls_acc for accuracy?
- I guess that X,Y,W,H are for a bounding box, but for a training image or a validation image? Can you tell me a little bit more about it?

Thank you very much,

Damien

**Comment by Flova** [comment]
Hey Damien,
sorry for the late answer. I just took over this repo.
* The validation dataset is used during training after each epoch to calculate the mAP values displayed for each class. You also get metrics for each batch but these are related to the training (loss etc.) and are estimated via the training dataset.
* These layers are the output layers of the Yolo network. It has multiple, outputs at different "depths" of the network. Multiscaling is a different thing. It says that the network is trained using multiple resolutions randomly. Because it only consists of convolutions it is able to handle different resolutions without changes to the topology. The grid size states the number of grid cells, which are a core component of the Yolo architecture. If the resolution changes the number of cells also changes.
* As I said multiscaling is a different thing than multiple output layers. If you uplead your `.cfg` to https://netron.app/ you see the graph and the output layers,
* `cls_acc` states the accuracy of the class prediction on the training set
* training labels and validation labels are identical. They follow the normal Yolo style with one box representing one line in the `.txt` file. There is a `.txt` for each image with the same name (but ending with `.txt` instead of e.g. `.png`). The labels are in `labels/` and the images in `images/`. In the `train.txt` are the paths of all images used for training. Similar for the `val.txt`. Both of them are referenced via a `.data` file. See `coco.data` for reference. Bounding boxes are noted in relative coordinates (fraction of the image width/height) in the style `ClassID X Y W H `

Flo

**Comment by Flova** [comment]
I hope this answers your question. Feel free to reopen for further discussion.
